core-war/src/main/webapp/map/js/services/silverpeas-map-user-service.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2000 - 2022 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//# sourceURL=/map/js/services/silverpeas-map-user-service.js
+
+(function() {
+
+  /**
+   * User service dedicated to map management.
+   * Each time a user identifier is fetched, it is put into a cache.
+   * @constructor
+   */
+  window.MapUserService = function() {
+    const __cache = {};
+    this.getById = function(userId) {
+      return new Promise(function(resolve) {
+        const _user = __cache['user_' + userId];
+        if (!_user) {
+          sp.log.debug("MapUserService - Ajax request to get user data from userId " + userId);
+          User.getExtended(userId).then(
+              function(user) {
+                sp.log.debug("MapUserService - Caching userId " + user.id);
+                __cache['user_' + user.id] = user;
+                resolve(user);
+              },
+              function(reason) {
+                sp.log.debug("MapUserService - Caching userId " + userId + " (but it does not exist)");
+                __cache['user_' + userId] = true;
+                resolve(reason);
+              });
+        } else {
+          sp.log.debug("MapUserService - Using user cache to get user data from userId " + userId);
+          resolve((typeof _user === 'object') ? _user : false);
+        }
+      });
+    };
+  };
+})();
```

Comment:
I don't understand the reason of this service. And it seams not to be used anywhere.

Reply:
It is some specific projects that impulsed the implementation of Silverpeas's MAP APIs.
The aim of these specific projects is to display functional data about different kind of entities.
User data must often be fetched in order to be displayed on the map. And same user data could be fetched several times into context of display of different entities. This service permits to provide user data with cache features which avoid performing several same user data HTTP requests for a same user.